prow/plugins/cat/cat_test.go

```diff
@@ -175,16 +177,81 @@ func TestFormat(t *testing.T) {
 
 // Medium integration test (depends on ability to open a TCP port)
 func TestHttpResponse(t *testing.T) {
-	img := "http://localhost?kind=url"
+	// create test cases for handling content length of images
+	contentLength := make(map[string]string)
+	contentLength["/cat.jpg"] = "717987"
+	contentLength["/bigcat.jpg"] = "12647753"
+
+	// fake server for images
+	ts2 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if s, ok := contentLength[r.URL.Path]; ok {
+			body := "binary image"
+			w.Header().Set("Content-Length", s)
+			io.WriteString(w, body)
+		} else {
+			t.Errorf("Cannot find content length for %s", r.URL.Path)
+		}
+	}))
+	defer ts2.Close()
+	// get the port to construct image urls
+	port := ts2.Listener.Addr().(*net.TCPAddr).Port
```

Comment:
You don't need to extract the port like this. `ts2.URL` provides the URL of the server in the form `http://<ip>:<port>` so below you can specify `img := ts2.URL + "/cat.jpg"`...